### ISSUE TITLE
Buffer out of bounds reading file

### ISSUE THREAD

**Issue by thoni56** [Borewit/music-metadata#381]
[**Bug description**
I get the following error message when reading a particular file:

```
(node:21408) UnhandledPromiseRejectionWarning: RangeError [ERR_BUFFER_OUT_OF_BOUNDS]: Attempt to write outside buffer bounds
    at boundsError (internal/buffer.js:51:11)
    at Buffer.readUInt8 (internal/buffer.js:141:5)
    at Object.get (webpack:///./node_modules/music-metadata/lib/id3v2/ID3v2Token.js?:94:21)
    at Function.readData (webpack:///./node_modules/music-metadata/lib/id3v2/FrameParser.js?:11:66)
    at Function.readFrameData (webpack:///./node_modules/music-metadata/lib/id3v2/ID3v2Parser.js?:90:46)
    at ID3v2Parser.parseMetadata (webpack:///./node_modules/music-metadata/lib/id3v2/ID3v2Parser.js?:183:40)
    at ID3v2Parser.parseId3Data (webpack:///./node_modules/music-metadata/lib/id3v2/ID3v2Parser.js?:139:32)
(node:21408) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:21408) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I suspect that the data in the file is somehow corrupt, and that it is tricky to catch all error cases, but anyhow...

**Expected behavior**
No error ;-)

**Audio file demonstrating the problem**
Audio file attached.

~~[music-metadata-boundserror.zip](https://github.com/Borewit/music-metadata/files/4003452/music-metadata-boundserror.zip)~~

**Comment by Borewit** [comment]
Hi @thoni56, thanks for reporting your issue.
I had no issues parsing the file. Works [online using audio-tag-analyzer](https://audio-tag-analyzer.netlify.com/) as well.

**Comment by thoni56** [comment]
Sorry, wrong file. Nice touch with the on-line analyzer! Here's the problematic 
one.

[music-metadata-boundserror.zip](https://github.com/Borewit/music-metadata/files/4006969/music-metadata-boundserror.zip)

**Comment by Borewit** [comment]
Caused by an empty [ID3v2.3 MCDI tag](http://id3.org/id3v2.3.0#sec4.5) value.

The error can easily be prevented.

**Comment by Borewit** [comment]
Fixed in [v5.4.2](https://github.com/Borewit/music-metadata/releases/tag/v5.4.2)
